printing/printer.mli

```diff
@@ -173,21 +173,24 @@ val pr_transparent_state   : transparent_state -> Pp.t
 
 val pr_goal                : goal sigma -> Pp.t
 
-(** [pr_subgoals ~pr_first pp sigma seeds shelf focus_stack unfocused goals]
+(** [pr_subgoals ~pr_first ~diffs pp sigma seeds shelf focus_stack unfocused goals]
    prints the goals of the list [goals] followed by the goals in
    [unfocused], in a short way (typically only the conclusion) except
-   for the first goal if [pr_first] is true. This function can be
-   replaced by another one by calling [set_printer_pr] (see below),
-   typically by plugin writers. The default printer prints only the
+   for the first goal if [pr_first] is true. Also, if [diffs] is not None
+   and [pr_first] is true, then use [diffs] as the printed form of the
+   first goal. This function prints only the
    focused goals unless the conrresponding option
    [enable_unfocused_goal_printing] is set. [seeds] is for printing
    dependent evars (mainly for emacs proof tree mode). *)
-val pr_subgoals            : ?pr_first:bool -> Pp.t option -> evar_map -> seeds:goal list -> shelf:goal list -> stack:int list -> unfocused:goal list -> goals:goal list -> Pp.t
+val pr_subgoals            : ?pr_first:bool -> ?diffs:Pp.t -> Pp.t option -> evar_map -> seeds:goal list -> shelf:goal list -> stack:int list
+                             -> unfocused: goal list -> goals:goal list -> Pp.t
 
 val pr_subgoal             : int -> evar_map -> goal list -> Pp.t
 val pr_concl               : int -> evar_map -> goal -> Pp.t
 
+val pr_open_subgoals2      : ?quiet:bool -> ?diffs:Pp.t -> proof:Proof.t -> unit -> Pp.t
```

Comment:
Why is there a unit?

Reply:
You can't have a function with only optional and labelled parameters.  The () is required to mark the last parameter.  It will not compile without it. Manual section 4.1.1:

````
A function taking some optional arguments must also take at least one non-optional argument.
The criterion for deciding whether an optional argument has been omitted is the non-labeled
application of an argument appearing after this optional argument in the function type.
```